src/ImageSearchModal.jsx

```diff
@@ -0,0 +1,134 @@
+import React from 'react';
+import { Button, ListGroup, ListGroupItem, Modal } from 'patternfly-react';
+
+import { ErrorNotification } from './Notification.jsx';
+import * as utils from './util.js';
+import varlink from './varlink.js';
+import cockpit from 'cockpit';
+
+import '../lib/form-layout.less';
+import './ImageSearchModal.css';
+
+const _ = cockpit.gettext;
+
+export class ImageSearchModal extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            imageIdentifier: '',
+            imageList: [],
+            searchInProgress: false,
+        };
+        this.onDownloadClicked = this.onDownloadClicked.bind(this);
+        this.onItemSelected = this.onItemSelected.bind(this);
+        this.onSearchTriggered = this.onSearchTriggered.bind(this);
+        this.onValueChanged = this.onValueChanged.bind(this);
+    }
+
+    componentDidMount() {
+        this._isMounted = true;
+    }
+
+    componentWillUnmount() {
+        this._isMounted = false;
+    }
+
+    onDownloadClicked() {
+        let selectedImageName = this.state.imageList[this.state.selected].name;
+
+        this.props.close();
+        this.props.downloadImage(selectedImageName);
+    }
+
+    onItemSelected(key) {
+        this.setState({ selected: key });
+    }
+
+    onSearchTriggered() {
+        if (!this._isMounted)
+            return;
+
+        varlink.call(utils.PODMAN_ADDRESS, "io.podman.SearchImage", { name: this.state.imageIdentifier })
+                .then(reply => this.setState({ imageList: reply.images || [], searchInProgress: false }))
+                .catch(ex => {
+                    this.setState({
+                        searchInProgress: false,
+                        dialogError: _("Failed to search for new images"),
+                        dialogErrorDetail: cockpit.format("$0: $1", ex.error, ex.parameters && ex.parameters.reason)
+                    });
+                });
+    }
+
+    onValueChanged(key, value) {
+        this.setState({ [key]: value });
+
+        if (key == 'imageIdentifier') {
+            this.setState({ searchInProgress: true });
```

Comment:
This makes the previous result list disappear as soon as I change the string, which can be quite jarring. The docker page requires an explicit `Return` key stroke.

Reply:
Keystrokes that trigger autocomplete searches should not include spaces... and we should show the results _possibly greyed out_ with a progress indicator (spinner) to show that new results are being fetched, instead of immediately clearing everything away.

Reply:
I fixed Lars's comment but I don't really know how to show results greyed out and in front of them a progress indicator. I 'll be glad to implement it however if you can provide some hints.